workers/unity/Assets/Gdk/Core/Worker/WorkerBase.cs

```diff
@@ -45,22 +41,47 @@ public bool Connect(ConnectionConfig config)
                 Connection = ConnectionUtility.LocatorConnectToSpatial((LocatorConfig) config, GetWorkerType);
             }
 
-            if (Connection == null)
+            if (Connection == null || !Connection.IsConnected)
             {
                 return false;
             }
 
+            World = new World(WorkerId);
+            WorkerRegistry.SetWorkerForWorld(this);
+
+            View = new MutableView(World);
             View.Connection = Connection;
 
-            Application.quitting += () =>
```

Comment:
This should stay here. Having `OnDisable` called in the bootstrap and the application calling `OnQuit` is not the same.

Reply:
cc @JonasImprobable had worked on a bug about this I think

Reply:
We need to clean up the world when the game object is disabled as well because if we don't there is a bunch of null refs caused by rigidbodies being null because they are no longer in the scene. We could also have Application.quitting call clean in addition to this though. Is there ever a situation where OnDisable isn't called but Application.quitting is?